src/main.rs

```diff
@@ -1,10 +1,35 @@
 use std::env;
+use std::io::{self, Write};
+use std::process;
+use std::borrow::Cow;
 
-fn main() {
-    let arguments = env::args().collect::<Vec<_>>();
-    
-    match arguments.len() {
-        1 => loop { println!("y") },
-        _ => loop { println!("{}", arguments[1]) },
+const BUFFER_CAPACITY: usize = 64 * 1024;
+
+fn write(output_bytes: &[u8]) {
+    let stdout = io::stdout();
+    let mut locked = stdout.lock();
+    let mut buffer = [0u8; BUFFER_CAPACITY];
+    let mut buffer_size = output_bytes.len();
+
+    let _ = (&mut buffer[..]).write(output_bytes);
+    while buffer_size < BUFFER_CAPACITY {
+        let (left, mut right) = buffer.split_at_mut(buffer_size);
+        if !locked.write_all(left).is_ok() {
+            return;
+        }
+        right.write(left).unwrap();
+        buffer_size *= 2;
     }
+
+    while locked.write_all(&buffer).is_ok() {}
+}
+
+fn main() {
+    write(env::args()
+        .skip(1)
+        .next()
```

Comment:
Yes, there is a clippy lint. And no, I was on mobile, so I didn't run it – I guess I sort of internalized it.